### PR TITLE
Addresses 721: added simple check for valid url, removed regex, and passed existing …

### LINKED ISSUE #721: Bug: Enhance Repository URL parsing before adding
The JS function to parse valid Github Repo URLs defined in the following needs to be enhanced for accurate parsing.

https://github.com/open-sauced/open-sauced/blob/master/src/lib/util.js

### Describe the bug
Proof of Concept - 
https://jsfiddle.net/sujitagarwal/p3yucL0z/

### Expected behavior
All and only valid Github Repo URL patterns must be accepted

### PR DESCRIPTION
**Pull request by CodeItQuick**
…tests

I think I'm good with it for now, although I could add more checks if necessary (like adding a list of false extensions for .org, .io, etc.)

## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [x] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description
Some simple refactoring to check that a valid url is being requested. Also got rid of the regex, I don't think its needed here?

Package-lock.json file is incredibly minor change, and can probably be ignored/removed. The gitignore was so you didn't get my wallaby-related config files.

## Related Tickets & Documents

Fixes #721 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

Option d - there were already tests written, I just refactored

## Added to documentation?

- [ ] 📜 readme
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?
I didn't try running the tests locally, but I assume they will run automatically upon opening the PR. "Didn't run the tests locally" actually means they didn't all run and pass locally but that's due to my current setup, I believe they all work (and the local ones that are concerned with this block of code work, and pass).

## [optional] What gif best describes this PR or how it makes you feel?

That cool parrot gif, or :/. I'm inbetween on it, if you need revisions or if my answer is to simple feedback is appreciated.

### CODE REVIEW

**Comment by bdougie** [comment]
Thanks for the PR @CodeItQuick. Real quick, you can check boxes by simply clicking them or adding [x]. I see you added [checked].

I also see the staging environment failed. Retrying that build.

**Comment by CodeItQuick** [comment]
The boxes are checked with x's, I'll have to dig into why the build failed, as I was hoping it was not my fault, by some weird math that doesn't make sense.

**Comment by CodeItQuick** [comment]
Thanks @vmitchell85 for the help! He helped PR this in on stream.

File: src/lib/util.js [hunk 1]
```diff
@@ -1,11 +1,50 @@
...
+    const [owner, repo] = relativeRepoUrl.split("/");
+
+    if (!url.includes("github.com"))
+      return [false, null];
+
+    if (!owner || !repo || !(urlObject.protocol === "http:" || urlObject.protocol === "https:")) {
+      return [false, null];
+    }
+    return [true, owner + "/" + repo];
+  } catch (failedToConstructURL) {
```
Comment (bdougie) [inline src/lib/util.js:47]: Good to see, I saw the `_` earleir and was going to ask about that.

**Review by bdougie — APPROVED** [review]

File: src/lib/util.js (hunk 1 above)
Comment (CodeItQuick) [reply src/lib/util.js:47]: I almost think it should be named newUrlThrewError, but failedToConstructURL seems fine as-is. Yeah I agree - the _ seemed like bad practice.

**Comment by CodeItQuick** [comment]
Note: I found a recent bug involving case sensitivity. Although the url validator should be able to correctly lowercase GITHUB.com entries, it should not lowercase the owner_name or repo_name as it is case sensitive on githubs end. I removed the initial filtering done at the start of the function, and put it into just where it is needed.

**Review by filiptronicek — APPROVED** [review]

**Pull request merged**